src/runtime/storage/storage-provider-base.ts

```diff
@@ -19,6 +19,30 @@ enum EventKind {
 }
 type Callback = ({}) => void;
 
+/**
+ * Methods that must be implemented by a Variable Storage Provider
+ */
+export interface VariableStorageProvider extends StorageProviderBase {
+  // tslint:disable-next-line: no-any
+  get(): Promise<any>;
+  set(val: {}): Promise<void>;
+}
+
+/**
+ * Methods that must be implemented by a Collection Storage Provider
+ */
+export interface CollectionStorageProvider extends StorageProviderBase {
+  // tslint:disable-next-line: no-any
+  toList(): Promise<any[]>;
+
+  // tslint:disable-next-line: no-any
+  getMultiple(ids: string[]): Promise<any[]>;
+  storeMultiple(values, keys: string[], originatorId: string): Promise<void>;
+
+  // tslint:disable-next-line: no-any
+  get(id: string): any;
```

Comment:
I think you can use {} instead of any, but then maybe this makes the call sites uglier.

Reply:
seems to work fine!  Thanks for the suggestion.

Reply:
I take that back :(   will find a better solution in a followup PR.